core/src/autogluon/core/trainer/abstract_trainer.py

```diff
@@ -1134,6 +1134,15 @@ def save(self):
         if self.low_memory:
             self.models = models
 
+    def compile_models(self, **kwargs):
+        models_to_compile = list(self.models.keys())
+        if len(models_to_compile) == 0:
+            raise RuntimeError("No valid persisted model to be compiled. "
+                               "Make sure make a call to `persist_models` before trying to compile the models.")
```

Comment:
The models should not need to be persisted for this.

Reply:
Deleted, persist_model() is no longer required before compile_model().